### PR TITLE
gh-103142: Upgrade binary builds and CI to OpenSSL 1.1.1u

### PR DESCRIPTION
**Pull request by gpshead**
This OpenSSL version addresses a pile if less-urgent CVEs since 1.1.1t.

The `Mac/BuildScript/build-installer.py` file appears to have already been updated.

* Issue: gh-103142

### CODE REVIEW

**Comment by gpshead** [comment]
#105129 does more of this, so anything missing can probably just be moved over into that. @ned-deily

**Comment by ned-deily** [comment]
@gpshead Sorry I didn't ping you earlier. I did not attempt to do the Windows changes. I figured that should be in a separate PR since the versions don't always match up. But feel free to use either PR.

**Comment by zooba** [comment]
I'm doing updated binaries for Windows now. I'll let you know when they're done

**Comment by zooba** [comment]
Binaries are up now. I'll hit rerun

File: Modules/_ssl_data_300.h
Comment (gpshead) [inline Modules/_ssl_data_300.h:1]: vs https://github.com/python/cpython/pull/105129/files I manually reverted edits that removed definitions from this file as for the purposes of backporting, I don't want names to disappear.  some were FIPS related and I'm blindly guessing that those may be special to some vendor openssl builds without digging into the history?  regardless there is no harm in keeping names, everything is #ifdef based.

File: Modules/_ssl_data_31.h
Comment (gpshead) [inline Modules/_ssl_data_31.h:1]: I'm adding this, generated from 3.1.1 but do not intend to backport it beyond 3.12 (and if it causes anyone trouble in 3.12-land, we can just revert it there or re-add whatever is missing).

diff it vs the _300 file, you'll see the things missing that I avoided removing from the _300 one.

**Comment by gpshead** [comment]
many backport labels removed as older backports will be generated from the 3.11 backport PR and successively chained ones as many backport edits will be the same.

**Review by ned-deily — APPROVED** [review]

**Pull request merged**